### PR TITLE
Add DDS communication Blackbox Test [13199]

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
Signed-off-by: Juan López Fernández <juanlopez@eprosima.com>

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -78,6 +78,7 @@ endif(WIN32)
 # TEST_NAME -> test name (name of the class of the test in Test .cpp)
 # TEST_LIST -> test cases implemented in the Test .cpp
 # TEST_EXTRA_LIBRARIES -> libraries that must be linked to compile the test
+# [TEST_EXTRA_HEADERS] -> a sixth optional argument with extra headers needed for the test
```
Comment (rsanchez15) [inline test/CMakeLists.txt:81]:
```suggestion
# ARGV5 -> extra headers needed for the test (sixth optional argument)
```

File: test/blackbox/CMakeLists.txt [hunk 2]
```diff
@@ -16,8 +16,8 @@
 # Arguments:
 # TEST_NAME -> test name (it will add "_Test" after name)
 # TEST_SOURCES -> sources for the test
-# TEST_HEADERS -> headers for the test
 # TEST_LIST -> test cases implemented in the Test .cpp
+# [TEST_EXTRA_HEADERS] -> a fifth optional argument with extra headers needed for the test
```
Comment (rsanchez15) [inline test/blackbox/CMakeLists.txt:20]:
```suggestion
# ARGV4 -> extra headers needed for the test (fifth optional argument)
```

File: test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+# This .yaml contains the configuration for the complex test.
```
Comment (rsanchez15) [inline test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml:2]:
```suggestion
# This YAML configures the DDS Router for the DDS communication test.
```

File: test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# It is composed of two simple (RTPS) participants.
```
Comment (rsanchez15) [inline test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml:3]:
```suggestion
# It is composed of two Simple Participants.
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: test/CMakeLists.txt (hunk 1 above)
Comment (juanlofer-eprosima) [reply test/CMakeLists.txt:81]: :+1:

File: test/blackbox/CMakeLists.txt (hunk 2 above)
Comment (juanlofer-eprosima) [reply test/blackbox/CMakeLists.txt:20]: :+1:

File: test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml (hunk 3 above)
Comment (juanlofer-eprosima) [reply test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml:2]: :+1:

File: test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml (hunk 4 above)
Comment (juanlofer-eprosima) [reply test/blackbox/ddsrouter_core/resources/dds_test_simple_configuration.yaml:3]: :+1:

**Review by rsanchez15 — DISMISSED** [review]
Approved considering that this will be refactored in order to be compatible with any DDS user participant (SIMPLE, CLIENT, SUPERCLIENT and SERVER)

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**